spec/services/existing_candidate_authentication_spec.rb

```diff
@@ -11,8 +11,26 @@
   end
 
   describe '#execute' do
+    context 'when the candidate already has 3 application choices' do
+      it 'sets the candidates course_from_find_id to nil and returns :candidate_already_has_3_courses' do
+        new_course = create(:course)
+        candidate = create(:candidate, course_from_find_id: new_course.id)
+        provider = create(:provider)
+        application_form = create(:application_form, candidate: candidate)
+        3.times { course_option_for_provider(provider: provider) }
+        provider.courses.each do |course|
+          create(:application_choice, application_form: application_form, course_option_id: course.course_options.first.id)
+        end
```

Comment:
The 3 courses don't have to be from the same provider, so perhaps these lines can be shortened to:

```
create(:completed_application_form, candidate: candidate, application_choices_count: 3)
```

Reply:
Nice. Have done. Thanks 👍